### PR TITLE
run zk scripts without rust code

### PR DESCRIPTION
**Pull request by freeranged3v**
# Description 

Running zk scripts is kind of painful. Need to write Rust to generate public inputs.

# Problem

It would be great to have a tool that can take zkscript and generate public inputs.

# Approach

Do a python binding (so it can be used for other things) and build zkas statement interpreter to generate public inputs with the binding.

### CODE REVIEW

**Comment by parazyd** [comment]
It would be useful to have a tool like zokrates. Maybe it could generate Rust code or something.

Would you be able to write a proof of concept?

**Comment by freeranged3v** [comment]
ah zokrates has a pretty comprehensive and easy to use set of commands. good source of inspiration. let me list out the commands and try implementing a couple.

**Comment by freeranged3v** [comment]
figuring from the halo2 book background material etc how to do the apis right.
pyo3 works well though!

**Comment by freeranged3v** [comment]
When you review, could you let me know if you think we should delete some python stuff and building on top of `zkas -e`? @parazyd

File: src/sdk-py/src/zk_binary.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+pub struct ZkBinary(pub(crate) decoder::ZkBinary);
+
+/// There is no need for constants, as bindings for ec_mul_short and ec_mul_base
+/// don't actually take the constants.
+/// The constants are hardcoded on the Rust side.
+#[pymethods]
+impl ZkBinary {
+    #[staticmethod]
+    fn decode(bytes: Vec<u8>) -> Self {
+        let bincode = decoder::ZkBinary::decode(bytes.as_slice()).unwrap();
```
Comment (freeranged3v) [inline src/sdk-py/src/zk_binary.rs:14]: Zkrunner uses `ZkBinary::decode`. Wdyt? Do you see more things from `zkas -e` we could use?

**Pull request closed without merging**